nifi-nar-bundles/nifi-standard-services/nifi-record-serialization-services-bundle/nifi-record-serialization-services/src/main/java/org/apache/nifi/schema/access/HortonworksEncodedSchemaReferenceStrategy.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.schema.access;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.schemaregistry.services.SchemaRegistry;
+import org.apache.nifi.serialization.record.RecordSchema;
+import org.apache.nifi.stream.io.StreamUtils;
+
+public class HortonworksEncodedSchemaReferenceStrategy implements SchemaAccessStrategy {
```

Comment:
I found the pattern used by looking at source code, since the Schema Registry is open source. I can look for documentation that spells this out. In my mind, I would steer clear of calling it something like IdentifierVersionSchemaReferenceStrategy for a few reasons. Firstly, it's quite wordy (though I guess mine is too) :) Secondly, the first byte of the pattern is a 'protocol' value that tells us how to interpret the rest of the bytes, how many bytes there are, etc. If the Hwx Schema Registry changes their serialization format - for instance, by adding new fields beyond identifier and version, or if they decide that the current format is too verbose and remove the version (assuming that a unique identifier would be given to each 'version') then this strategy can be updated to take advantage of that. If we went the route of IdentifierVersionSchemaReferenceStrategy, then we would need yet another strategy and it may be difficult for the user to know which one they need. Finally, it is not obvious from such a name why it would be used whereas a name like HortonworksEncodedSchemaReferenceStrategy makes it clear to users of the Hwx Schema Registry, I think, that this is compatible with the serializers & deserializers available there.